ArkhamOverlay/Pages/Overlay/OverlayController.cs

```diff
@@ -95,27 +90,66 @@ public class OverlayController : Controller<OverlayView, OverlayViewModel> {
                 overlayCards.CollectionChanged += (s, e) => CalculateMaxHeightForCards();
             }
 
-            foreach (var selectableCards in appData.Game.AllSelectableCards) {
-                InitializeSelectableCards(selectableCards);
-            }
-
             View.Closed += (s, e) => {
-                eventBus.UnsubscribeFromShowDeckListRequest(ShowDeckListRequest);
-                eventBus.UnsubscribeFromTakeSnapshotRequest(TakeSnapshot);
-
                 _autoSnapshotTimer.Stop();
-                foreach (var overlayCards in ViewModel.AllOverlayCards) {
-                    foreach (var overlayCard in overlayCards) {
-                        overlayCard.Card.IsDisplayedOnOverlay = false;
-                    }
-                }
+                ClearAllCards();
+
+                eventBus.UnsubscribeFromButtonInfoChanged(ButtonInfoChangedHandler);
+                eventBus.UnsubscribeFromButtonRemoved(ButtonRemovedHandler);
+                eventBus.UnsubscribeFromShowDeckListRequest(ShowDeckListRequestHandler);
+                eventBus.UnsubscribeFromTakeSnapshotRequest(TakeSnapshotHandler);
+                eventBus.UnsubscribeFromToggleCardInfoVisibilityRequest(ToggleCardInfoVisibilityRequestHandler);
+                eventBus.UnsubscribeFromClearAllCardsRequest(ClearAllCardsRequestHandler);
+                eventBus.UnsubscribeFromClearAllCardsForCardGroupRequest(ClearAllCardsForCardGroupRequestHandler);
+                eventBus.UnsubscribeFromToggleCardZoneVisibilityRequest(ToggleCardZoneVisibilityRequestHandler);
 
                 Closed?.Invoke();
             };
 
             _logger.LogMessage("Finished initializing Overlay Controller.");
         }
 
+        public double Top { get => View.Top; set => View.Top = value; }
+        public double Left { get => View.Left; set => View.Left = value; }
+
+        public void Close() {
+            _logger.LogMessage("Closing overlay window.");
+            View.Close();
+        }
+
+        public void Activate() {
+            _logger.LogMessage("Activating overlay window.");
+            View.Activate();
+        }
+
+        internal void Show() {
+            _logger.LogMessage("Showing overlay window.");
+            View.Show();
+        }
+
+        public event Action Closed;
+
+        private ICardZoneManager CreateCardZoneManager(IList<OverlayCardViewModel> overlayCards) {
+            var cardZoneManager = ServiceLocator.GetService<CardZoneManager>();
```

Comment:
Will this just return the same instance every time? Or will it new up a CardZoneManager for every call?

Reply:
It will new up a new one for every call- which is the intent, as this managger tracks information specific for the card zone in question (top or bottom).